### ISSUE TITLE
java_import not made available at runtime

### ISSUE THREAD

**Issue by guw** [bazelbuild/bazel#15111]
### Description of the problem / feature request:

```
java_binary(
    name = "mybuilder1",
    main_class = "some.MyClass"
    runtime_deps = [
        ...
    	":resourcejar1",
    ],
    ...
)

java_binary(
    name = "mybuilder2",
    main_class = "some.MyClass"
    runtime_deps = [
        ...
    	":resourcejar2",
    ],
    ...
)


# create the file set for packaging
pkg_files(
    name = "resourcejar1_files",
    srcs = glob(include = ["resources/**"]),
    # drop everything from the tree up until the directory itself
    strip_prefix = strip_prefix.from_pkg("resources"),
    # where it should be in the final package
    prefix = None,
)

# create the jar for java_import
pkg_zip(
    name ="resourcejar1_jar",
    srcs = [":resourcejar1_files"],
    package_file_name = "resourcejar1.jar",
)

# make it available to java_library
java_import(
    name = "resourcejar1",
    jars = [":resourcejar1_jar"],
)

java_library(
    name = "resourcejar2",
    resources = glob(include = ["resources/**"]),
    resource_strip_prefix = "resources",
)

```
`bazel run :mybuilder1` does not work - resources missing at runtime
`bazel run :mybuilder2` works as expected

### Feature requests: what underlying problem are you trying to solve with this feature?

`java_library` does not allow re-mapping. We have some more complicated cases which require use of `prefix` available in `pkg_files`. In Maven there is target path for this, which we are trying to map.

### Steps to reproduce

A more complete repo is available at:
```
git clone [git@github.com](mailto:git@github.com):salesforce/bazel-java-builder-template.git
cd bazel-java-builder-template/examples/001-simple
bazel build :all
```

The build fails as soon as you revert https://github.com/salesforce/bazel-java-builder-template/commit/2fb1741ed2abc78d74453bdfbb6875222a1ab93c

**Comment by cushon** [comment]
> bazel run :mybuilder1 does not work - resources missing at runtime
> bazel run :mybuilder2 works as expected

I cannot reproduce. Can you provide a self-contained minimal repro?

```
=== BUILD
load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
load("@rules_pkg//:pkg.bzl", "pkg_zip")

java_binary(
    name = "mybuilder1",
    srcs = ["some/MyClass.java"],
    main_class = "some.MyClass",
    runtime_deps = [
        ":resourcejar1",
    ],
)

java_binary(
    name = "mybuilder2",
    srcs = ["some/MyClass.java"],
    main_class = "some.MyClass",
    runtime_deps = [
        ":resourcejar2",
    ],
)

# create the file set for packaging
pkg_files(
    name = "resourcejar1_files",
    srcs = glob(include = ["resources/**"]),
    # where it should be in the final package
    prefix = None,
    # drop everything from the tree up until the directory itself
    strip_prefix = strip_prefix.from_pkg("resources"),
)

# create the jar for java_import
pkg_zip(
    name = "resourcejar1_jar",
    srcs = [":resourcejar1_files"],
    package_file_name = "resourcejar1.jar",
)

# make it available to java_library
java_import(
    name = "resourcejar1",
    jars = [":resourcejar1_jar"],
)

java_library(
    name = "resourcejar2",
    resource_strip_prefix = "resources",
    resources = glob(include = ["resources/**"]),
)
=== resources/bar.txt
bar
=== resources/foo.txt
hello
=== some/MyClass.java
package some;

import java.net.URL;

public class MyClass {
  public static void main(String[] args) {
    URL foo = MyClass.class.getResource("/foo.txt");
    System.err.println(foo);
  }
}
=== WORKSPACE
workspace(name = "test")

load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

# development time dependencies
http_archive(
    name = "rules_pkg",
    sha256 = "62eeb544ff1ef41d786e329e1536c1d541bb9bcad27ae984d57f18f314018e66",
    urls = [
        "https://mirror.bazel.build/github.com/bazelbuild/rules_pkg/releases/download/0.6.0/rules_pkg-0.6.0.tar.gz",
        "https://github.com/bazelbuild/rules_pkg/releases/download/0.6.0/rules_pkg-0.6.0.tar.gz",
    ],
)

load("@rules_pkg//:deps.bzl", "rules_pkg_dependencies")

rules_pkg_dependencies()
```

```
$ bazel run :mybuilder1
jar:file:bazel-out/k8-fastbuild/bin/mybuilder1.runfiles/test/resourcejar1.jar!/foo.txt
$ bazel run :mybuilder2
jar:file:bazel-out/k8-fastbuild/bin/mybuilder2.runfiles/test/libresourcejar2.jar!/foo.txt
```

**Comment by guw** [comment]
I did further investigation. 

I noticed that `getClass().getResource("/files/resource2.txt")` *always* works. What is actually broken is Springs' resource resolver, which allows scanning for `classpath*:/files/*.txt`. This is broken. At this point I believe it's a JDK/Spring issue. Thus, I opened https://github.com/spring-projects/spring-framework/issues/28230.

I noticed is the following: The jar file produced with `pkg_files` + `pkg_zip` does not contain a manifest file. This is the only difference. This seems to break it somehow. But note that Spring's scanning is also broken in the fat jar, which does have a manifest.
